recipes/libtorch/all/conanfile.py

```diff
@@ -0,0 +1,667 @@
+        # - gloo
+        # - tensorpipe
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+        if self.options.get_safe("with_numa") and not self.dependencies["libnuma"].options.shared:
+            raise ConanInvalidConfiguration(
+                "libtorch requires libnuma shared. Set '-o libnuma/*:shared=True', or disable numa with "
+                "' -o libtorch/*:with_numa=False'"
+            )
+
+        if self.options.blas != "openblas":
+            # FIXME: add an independent LAPACK package to CCI
+            raise ConanInvalidConfiguration("'-o libtorch/*:blas=openblas' is currently required for LAPACK support")
+
+        if self.options.blas == "veclib" and not is_apple_os(self):
+            raise ConanInvalidConfiguration("veclib only available on Apple family OS")
+
+        if self.options.get_safe("with_cuda"):
+            self.output.warning("cuda recipe is not available, assuming that NVIDIA CUDA SDK is installed on your system")
+            if self.options.get_safe("with_cudnn"):
+                self.output.warning("cudnn recipe is not available, assuming that NVIDIA CuDNN is installed on your system")
+            if self.options.get_safe("with_nvrtc"):
+                self.output.warning("nvrtc recipe is not available, assuming that NVIDIA NVRTC is installed on your system")
+            if self.options.get_safe("with_cusparselt"):
+                self.output.warning("cusparselt recipe is not available, assuming that NVIDIA cuSPARSELt is installed on your system")
+        if self.options.get_safe("with_nccl"):
+            self.output.warning("nccl recipe is not available, assuming that NVIDIA NCCL is installed on your system")
+        if self.options.get_safe("with_rocm"):
+            self.output.warning("rocm recipe is not available, assuming that ROCm is installed on your system")
+        if self.options.get_safe("with_xpu"):
+            self.output.warning("xpu recipe is not available, assuming that Intel oneAPI is installed on your system")
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.18 <4]")
+        self.tool_requires("cpython/[~3.12]")
+        if self._depends_on_flatbuffers:
+            self.tool_requires("flatbuffers/<host_version>")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+        # Keep only a restricted set of vendored dependencies.
+        # Do it before build() to limit the amount of files to copy.
+        allowed = ["pocketfft", "kineto", "miniz-2.1.0"]
```

Comment:
For MacOs build some more 3rd party libs are neeeded
```suggestion
        allowed = ["pocketfft", "kineto", "miniz-2.1.0"]
        if self.is_mac_os == True:
            allowed = ["pocketfft", "kineto", "miniz-2.1.0", "opentelemetry-cpp", "protobuf"]
```

Reply:
Hmm... I would prefer to unvendor these and use Conan versions. I'll have to do more testing on macOS.

Reply:
I total agree with you! It's a fast workaround.